src/syntax/ast.rs

```diff
@@ -1,40 +1,67 @@
 #[derive(Debug, PartialEq)]
-pub struct Resource(pub Vec<Entry>);
+pub struct Resource {
+    pub body: Vec<Entry>,
+}
 
 #[derive(Debug, PartialEq)]
 pub enum Entry {
     Message(Message),
 }
 
 #[derive(Debug, PartialEq)]
-pub struct Identifier(String);
-
-#[derive(Debug, PartialEq)]
 pub struct Message {
-    pub id: String,
+    pub id: Identifier,
     pub value: Option<Pattern>,
     pub traits: Option<Vec<Member>>,
 }
 
 #[derive(Debug, PartialEq)]
 pub struct Pattern {
     pub elements: Vec<PatternElement>,
+    pub quoted: bool,
 }
 
 #[derive(Debug, PartialEq)]
 pub enum PatternElement {
     Text(String),
-    Placeable(Vec<Expression>),
+    Placeable { expression: Vec<Expression> },
```

Comment:
Maybe call it `expressions` (plural) until we make the syntax changes discussed in tools-l10n?